src/components/signup/subcomponents/SignUpForm.js

```diff
@@ -0,0 +1,90 @@
+import React, {Component} from 'react';
+import {Input, Icon, Row, Card, Button, ProgressBar, Col} from 'react-materialize';
+import styles from '../../../styles/signup.scss';
+import PropTypes from 'prop-types';
+import {notify} from "react-notify-toast";
+import Notifications from "react-notify-toast";
+
+class SignUpForm extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+                username: '',
+                email:'',
+                password:'',
+                confirm_password:'',
+                visible: false,
+
+        };
+
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    componentWillReceiveProps(receivedProp){
+        console.log('all received props',receivedProp)
+        if(!receivedProp.visible){
+            this.setState({visible: false})
+        }
+    }
+
+    handleChange(event) {
+
+        const {name,value}  = event.target;
+        this.setState({[name]: value});
+
+    }
+
+    handleSubmit(event) {
+
+        event.preventDefault();
+
+         this.setState({visible: true});
+
+        if(this.state.password.trim() === this.state.confirm_password.trim()) {
+
+             this.props.userSignUpRequest(this.state);
+        }
+
+        else{
+            notify.show('Passwords don\'t match please', 'error',4000)
+            this.setState({visible: false})
+
+        }
+
+    }
+
+    render() {
+        const isVisible = this.state.visible;
+        const type ='password'
+        return (
+            <div>
+                <Notifications/>
+                    {isVisible && <ProgressBar id='progress_Bar' name='progress_Bar'/>}
+                    <form className={styles["card-form-signup"]} onSubmit={this.handleSubmit}>
```

Comment:
We would make this file much neater if the <form>...</form> code was in its own component -- A stateless component